### ISSUE TITLE
Array of shorts is detected as integerLiteral by AST matcher

### ISSUE THREAD

**Issue by Zilleplus** [llvm/llvm-project#59236]
LLVM built at commit  609b95cea16ea18330ca4f4ec2edb60114cc69cf (latest main)
Reproduced with:
 
```console
$ clang-query test.cpp
clang-query> match integerLiteral(hasType(asString("long")))
```
result:
```console
Match #1:

/tmp/test.cpp:4:16: note: "root" binds here
  for(auto A : Accs) 
               ^~~~
```

With `test.cpp`:
```c++
const static short Accs[] = {}; 
void foo(){
  for(auto A : Accs) 
  {}
}
```

It doesn't seem logical that a integerLiteral would match on a short array, or am I missing something?

**Comment by Zilleplus** [comment]
@rymiel thank you for the quick and clear answer. So this is clearly not a bug but a feature, I will close the issue.
